Update serving docs for LLM flat dict input (#11189)
mlflow/mlflow: docs/source/deployment/deploy-model-locally.rst

JSON Input
**********

You can either pass a flat dictionary corresponding to the desired model payload or wrap the
payload in a dict with a dict key that specifies your payload format. 

> Raw Payload Dict

Reviewer: Can we change the order to start from the traditional wrapped input, and mention that this is special format introduced for providing compatible interface with SaaS APIs? Most of the models except OpenAI chat still need the original input format, so probably we still want to say it is "normal" and raw dict is a sort of backdoor.

^^^^^^^^^^^^^^^^

If your payload is in a format that your mlflow served model will accept and it's in the supported
models below, you can pass a raw payload dict.

.. list-table::
    :widths: 20 40 40
    :header-rows: 1
    :class: wrap-table

    * - Supported Model Type
      - Description
      - Example

>     * - OpenAI Chat

Reviewer: I'm afraid that "Model Type" is a bit vague concept (like it can be easily mistaken with flavor). Since we indeed have no limitation for models - it's just matter of customer's model accepts the Chat payload or not - probably we can replace it with sth like "supported request format"?

Author: Yeah this is tough because it feels like a one-off. Good clarification

      - `OpenAI chat request payload <https://platform.openai.com/docs/api-reference/chat/create>`_.†
      - 
        .. code-block:: python

>           {"messages": [{"role": "user", "content": "Tell a joke!"}], "temperature": 0.0}

Reviewer: nit: could you format this with indentation? (Table view is not truncated so has relatively low durability for line length)
```
{
    "messages": [
        {
            "role": "user",
            "content": ...
```

Author: Linting reformats. I noqa-ed, but for ruff I don't think there's a convenient noqa range operator. Instead of commenting `# noqa` on each line, I reformatted to the current commit. Lmk if I should add more to better align with your comment

Author: Rendered docs fit on my browser now

Reviewer: yea separating params is good enough, thx!

† Note that the ``model`` argument **should not** be included when using the OpenAI APIs, due to its configuration being set by the MLflow model instance. All other parameters can be freely used, provided that they are defined within the ``params`` argument within the logged model signature.

.. code-block:: python
  :caption: Example

>   # Prerequisite: serve an OpenAI model on localhost:5678 that defines

Reviewer: ```suggestion
  # Prerequisite: serve a Pyfunc model accepts OpenAI-compatible chat requests on localhost:5678 that defines
```

(to clearly differ it from openai flavor as discussed offline)

JSON Input
**********

You can either pass a flat dictionary corresponding to the desired model payload or wrap the
payload in a dict with a dict key that specifies your payload format. 

Raw Payload Dict
^^^^^^^^^^^^^^^^

If your payload is in a format that your mlflow served model will accept and it's in the supported
models below, you can pass a raw payload dict.

.. list-table::
    :widths: 20 40 40
    :header-rows: 1
    :class: wrap-table

    * - Supported Model Type
      - Description
      - Example

>     * - OpenAI Chat
>       - `OpenAI chat request payload <https://platform.openai.com/docs/api-reference/chat/create>`_.
>         Note that openai's ``model`` kwarg should not be included because we are querying the
>         mlflow served model. However, any other openai.chat parameter defined in our served model's 
>         signature will be applied.
>       - 
>         .. code-block:: python
> 
>           {"messages": [{"role": "user", "content": "Tell a joke!"}], "temperature": 0.0}
> 
> .. code-block:: python

Reviewer 2: ```suggestion
    * - OpenAI Chat
      - `OpenAI chat request payload <https://platform.openai.com/docs/api-reference/chat/create>`_.†
      - 
        .. code-block:: python

          {"messages": [{"role": "user", "content": "Tell a joke!"}], "temperature": 0.0}

† Note that the ``model`` argument **should not** be included when using the OpenAI APIs, due to its configuration being set by the MLflow model instance. All other parameters can be freely used, provided that they are defined within the ``params`` argument within the logged model signature.

.. code-block:: python
```

I think the table might look a bit cleaner if we add this note as a local footnote. WDYT?

Author: Looks great

  :caption: Example

  # Prerequisite: serve an OpenAI model on localhost:5678 that defines

>   #   `temperature` and `max_tokens` as parameters in our model signature

Reviewer 2: ```suggestion
  #   `temperature` and `max_tokens` as parameters within the logged model signature
```
